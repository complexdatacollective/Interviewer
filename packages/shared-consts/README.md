## @codaco/shared-consts

This repository exports shared constants that are used across the Network Canvas software.
