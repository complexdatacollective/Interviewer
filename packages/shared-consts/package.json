{
  "name": "@codaco/shared-consts",
  "version": "0.0.1-alpha.3",
  "description": "Shared constants and typescript definitions for the Network Canvas project.",
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "type": "module",
  "scripts": {
    "test": "xv dist",
    "lint": "xo",
    "start": "esno src/index.ts",
    "dev": "tsup --watch",
    "build": "tsup"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/complexdatacollective/shared-consts.git"
  },
  "keywords": [
    "typescript",
    "library"
  ],
  "author": "codaco",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/complexdatacollective/shared-consts/issues"
  },
  "homepage": "https://github.com/complexdatacollective/shared-consts#readme",
  "devDependencies": {
    "@tsconfig/node18": "^1.0.1",
    "@types/node": "^18.13.0",
    "esno": "^0.12.1",
    "tsup": "^6.6.0",
    "typescript": "^4.9.5",
    "xo": "0.46.4",
    "xv": "^1.0.2"
  },
  "files": [
    "dist",
    "!dist/**/*.test.*",
    "!dist/**/*.test.*.map",
    "!dist/**/*.test.d.ts"
  ],
  "publishConfig": {
    "access": "public"
  }
}
