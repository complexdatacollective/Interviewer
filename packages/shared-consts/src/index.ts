export * from './variables';
export * from './controls';
export * from './colors';
export * from './export-process';
export * from './network';
export * from './assets';
export * from './protocol';
export * from './session';
export * from './stages';

