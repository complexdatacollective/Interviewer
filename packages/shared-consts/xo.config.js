module.exports = {
  space: true,
  ignores: [
    '*.config.js',
  ],
};
