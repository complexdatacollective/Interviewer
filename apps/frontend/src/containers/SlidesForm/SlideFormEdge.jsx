import React, { PureComponent } from 'react';
import { find } from 'lodash';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withProps, compose } from 'recompose';
import { Scroller } from '@codaco/ui';
import { entityAttributesProperty, entityPrimaryKeyProperty } from '@codaco/shared-consts';
import { makeGetEdgeColor, getNetworkNodes } from '../../selectors/network';
import Node from '../Node';
import Form from '../Form';

class SlideFormEdge extends PureComponent {
  handleSubmit = (formData) => {
    const { id, onUpdate } = this.props;
    onUpdate(id, {}, formData);
  }

  render() {
    const {
      form,
      edgeColor,
      fromNode,
      toNode,
      subject,
      initialValues,
      submitButton,
      id,
      otherNetworkEntities,
      onScroll,
    } = this.props;

    return (
      <div className="swiper-slide">
        <div className="slide-content">
          <Node {...fromNode} />
          <div className="fake-edge" style={{ backgroundColor: `var(--${edgeColor})` }} />
          <Node {...toNode} />
          <div className="alter-form__form-container alter-edge-form__form-container">
            <Scroller onScroll={onScroll}>
              <Form
                {...form}
                className="alter-form__form alter-edge-form__form"
                initialValues={initialValues}
                autoFocus={false}
                subject={subject}
                onSubmit={this.handleSubmit}
                submitButton={submitButton}
                validationMeta={{ entityId: id }}
                otherNetworkEntities={otherNetworkEntities}
              />
            </Scroller>
          </div>
        </div>
      </div>
    );
  }
}

SlideFormEdge.propTypes = {
  form: PropTypes.object.isRequired,
  onUpdate: PropTypes.func.isRequired,
  onScroll: PropTypes.func,
  subject: PropTypes.object.isRequired,
  item: PropTypes.object.isRequired,
  submitButton: PropTypes.object,
};

SlideFormEdge.defaultProps = {
  onScroll: () => { },
  submitButton: <button type="submit" key="submit" aria-label="Submit" hidden />,
};

const withEdgeProps = withProps(({ item }) => ({
  id: item[entityPrimaryKeyProperty],
  initialValues: item[entityAttributesProperty],
}));

const withStore = connect((state, props) => {
  const getEdgeColor = makeGetEdgeColor();
  const nodes = getNetworkNodes(state);

  const fromNode = find(nodes, [entityPrimaryKeyProperty, props.item.from]);
  const toNode = find(nodes, [entityPrimaryKeyProperty, props.item.to]);

  return {
    fromNode,
    toNode,
    edgeColor: getEdgeColor(state, props.item),
  };
});

const EnhancedSlideFormEdge = withEdgeProps(SlideFormEdge);

export { EnhancedSlideFormEdge as SlideFormEdge };

export default compose(withStore, withEdgeProps)(SlideFormEdge);
