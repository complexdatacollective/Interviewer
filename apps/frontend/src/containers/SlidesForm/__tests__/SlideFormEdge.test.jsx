/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { entityAttributesProperty } from '@codaco/shared-consts';
import { SlideFormEdge } from '../SlideFormEdge';

const mockProps = {
  item: {
    to: 2,
    from: 1,
  },
  edgeColor: 'blue',
  nodes: {
    1: { name: 'One' },
    2: { name: 'Two' },
  },
  form: {
    title: 'alpha',
    entity: 'edge',
    type: 'friend',
  },
  edgeIndex: 1,
  stageIndex: 1,
};

const mockItem = {
  [entityAttributesProperty]: {
    foo: 'bar',
  },
};

describe('<SlideFormEdge />', () => {
  it('should render', () => {
    const component = shallow(<SlideFormEdge {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with prepopulated fields if provided', () => {
    const subject = shallow(<SlideFormEdge {...mockProps} item={mockItem} />);
    expect(
      subject.prop('initialValues'),
    ).toEqual({ foo: 'bar' });
  });
});
