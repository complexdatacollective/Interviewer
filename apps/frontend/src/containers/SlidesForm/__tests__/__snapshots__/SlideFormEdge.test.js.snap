// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideFormEdge /> should render 1`] = `
<SlideFormEdge
  edgeColor="blue"
  edgeIndex={1}
  form={
    Object {
      "entity": "edge",
      "title": "alpha",
      "type": "friend",
    }
  }
  item={
    Object {
      "from": 1,
      "to": 2,
    }
  }
  nodes={
    Object {
      "1": Object {
        "name": "One",
      },
      "2": Object {
        "name": "Two",
      },
    }
  }
  onScroll={[Function]}
  stageIndex={1}
  submitButton={
    <button
      aria-label="Submit"
      hidden={true}
      type="submit"
    />
  }
/>
`;
